src/Microsoft.AspNetCore.Razor.OmniSharpPlugin/DefaultProjectEngineFactory.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using System.Reflection;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+
+namespace Microsoft.AspNetCore.Razor.OmniSharpPlugin
+{
+    [Shared]
+    [Export(typeof(ProjectEngineFactory))]
+    internal class DefaultProjectEngineFactory : ProjectEngineFactory
+    {
+        private readonly object InitializationLock = new object();
+        private object _defaultEngineFactory;
+        private MethodInfo _createMethod;
+
+        public override RazorProjectEngine Create(
+            RazorConfiguration configuration,
+            RazorProjectFileSystem fileSystem,
+            Action<RazorProjectEngineBuilder> configure)
+        {
+            lock (InitializationLock)
+            {
+                if (_defaultEngineFactory == null)
+                {
+                    var razorWorkspaces = Assembly.Load("Microsoft.CodeAnalysis.Razor.Workspaces");
```

Comment:
In this assembly, all this reflection loading magic is needed due to strong naming restrictions 😢. The OmniSharpPlugin dll can't be strong named because OmniSharp isn't. Therefore, we can't IVT to it from Razor.